File: anddes-onboarding-backend/src/main/java/pe/compendio/myandess/onboarding/entity/Service.java
package pe.compendio.myandess.onboarding.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "service")
public class Service {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String description;

    private String icon;

    private String iconWeb;

    @OneToMany(mappedBy = "service", cascade = CascadeType.ALL)
    @OrderBy("id")
    private List<ServiceDetail> details;

    private Integer position;
}

File: anddes-onboarding-mobile-frontend/src/app/entity/service.ts
export interface Service {
    id : number;
    name : string;
    description :string;
    icon : string;
    iconWeb : string;
    position: number;
    details : ServiceDetail[];
}
export interface ServiceDetail{
    id : number;
    title : string;
    description : string;
    hidden : boolean;
    icon : string;
}
